File: adminweb/src/app/classes/accounts/AccountPenaltySetting.ts
import { PenaltyCalculationMethod } from "../products/PenaltySetting";
import { CurrencyLimitValue } from "../products/ValueConstraint";

export class AccountPenaltySetting {
    sameConstraintForAllCurrency = true;

    calculationMethod: PenaltyCalculationMethod = PenaltyCalculationMethod.OVERDUE_BALANCE;
    /**
     * Penalty Tolerance Period X Days.
     */
    penaltyTolerancePeriod = 0;
    /**
     * Penalty Rate Constraints (%).
     */
    penaltyRateValues: CurrencyLimitValue[] = [];
}

File: common-account/src/main/java/tech/corefinance/account/common/model/AccountPenaltySetting.java
package tech.corefinance.account.common.model;

import lombok.Data;
import tech.corefinance.product.common.enums.PenaltyCalculationMethod;
import tech.corefinance.product.common.model.CurrencyValue;
import tech.corefinance.product.common.model.ValueConstraint;

import java.util.List;

/**
 * PenaltySetting<br/>
 * How is the penalty rate charged? => % per year.
 */
@Data
public class AccountPenaltySetting {

    private boolean sameConstraintForAllCurrency;

    private PenaltyCalculationMethod calculationMethod;
    /**
     * Penalty Tolerance Period X Days.
     */
    private Integer penaltyTolerancePeriod;
    /**
     * Penalty Rate Constraints (%).
     */
    private List<CurrencyValue> penaltyRateValues;
}
